pkg/api/handlers/mcpgateway/oauth/authorize.go

```diff
@@ -341,7 +341,7 @@ func (h *handler) callback(req api.Context) error {
 
 // oauthCallback handles the second-level third-party OAuth for MCP servers.
 func (h *handler) oauthCallback(req api.Context) error {
-	oauthAuthRequestID, err := h.oauthChecker.stateCache.createToken(req.Context(), req.URL.Query().Get("state"), req.URL.Query().Get("code"), req.URL.Query().Get("error"), req.URL.Query().Get("error_description"))
+	oauthAuthRequestID, mcpID, err := h.oauthChecker.stateCache.createToken(req.Context(), req.URL.Query().Get("state"), req.URL.Query().Get("code"), req.URL.Query().Get("error"), req.URL.Query().Get("error_description"))
```

Comment:
nano-nit: I know you didn't name this variable, but it is really an `mcpServerID`. Using `mcpID` implicitly means that it could be an MCP server ID, an MCP server instance ID, or an MCP catalog entry ID.

If it is straightforward, we should rename this variable to `mcpServerID` here and where it is referenced in the session storage.